python-sdk/src/astro/databases/base.py

```diff
@@ -276,6 +276,61 @@ def drop_table(self, table: Table) -> None:
     # Table load methods
     # ---------------------------------------------------------
 
+    def create_schema_table_if_needed(
+        self,
+        table: Table,
+        file: File,
+        normalize_config: dict | None = None,
+        columns_names_capitalization: ColumnCapitalization = "original",
+        if_exists: LoadExistStrategy = "replace",
+        use_native_support: bool = True,
+    ):
+        """
+        Checks if the autodetect schema exists for native support else creates the schema and table
+        :param table: Table to create
+        :param file: File path and conn_id for object stores
+        :param normalize_config: pandas json_normalize params config
+        :param columns_names_capitalization:  determines whether to convert all columns to lowercase/uppercase
+        :param if_exists:  Overwrite file if exists
+        :param use_native_support: Use native support for data transfer if available on the destination
+        """
+        is_schema_autodetection_supported = (
+            self.check_schema_autodetection_is_supported(source_file=file)
+        )
+        is_file_pattern_based_schema_autodetection_supported = (
+            self.check_file_pattern_based_schema_autodetection_is_supported(
+                source_file=file
+            )
+        )
+
+        if (
+            use_native_support
+            and is_schema_autodetection_supported
+            and not file.is_pattern()
+        ):
+            return
+        if (
+            use_native_support
+            and file.is_pattern()
+            and is_schema_autodetection_supported
+            and is_file_pattern_based_schema_autodetection_supported
+        ):
+            return
+
+        self.create_schema_if_needed(table.metadata.schema)
+        if if_exists == "replace" or not self.table_exists(table):
+            self.drop_table(table)
+            self.create_table(
+                table,
+                resolve_file_path_pattern(
```

Comment:
@utkarsharma2 result wont be empty as it will raise value error in https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/files/base.py#L155